gcp/apis/servicenetworking/v1alpha1/connection_types.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/crossplaneio/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// ConnectionSpec defines the desired state of a Connection
+type ConnectionSpec struct {
+	v1alpha1.ResourceSpec `json:",inline"`
+	ConnectionParameters  `json:",inline"`
+}
+
+// ConnectionParameters specifies the configuration of a Connection.
+type ConnectionParameters struct {
+	// Parent: The service that is managing peering connectivity for a service
+	// producer's organization. For Google services that support this
+	// functionality, this value is services/servicenetworking.googleapis.com.
+	Parent string `json:"parent"`
+
+	// Network: The name of service consumer's VPC network that's connected
+	// with service producer network, in the following format:
+	// `projects/{project}/global/networks/{network}`.
+	// `{project}` is a project number, such as in `12345` that includes
+	// the VPC service consumer's VPC network. `{network}` is the name of
+	// the service consumer's VPC network.
+	Network string `json:"network"`
+
+	// ReservedPeeringRanges: The name of one or more allocated IP address
+	// ranges for this service producer of type `PEERING`.
+	ReservedPeeringRanges []string `json:"reservedPeeringRanges"`
+}
+
+// ConnectionStatus reflects the state of a Connection
+type ConnectionStatus struct {
```

Comment:
Is having `Peering` populated enough to declare that the connection is ready for use? I guess GCP doesn't provide a `State` information for this resource.

Reply:
As far as I can tell the connection API is basically an extension of the VPC peering API. Typically both VPCs being peered would need to configure a peering. The connection API configures peering in our project, and seems to automatically confirm the peering in a Google-controlled project. It's a pretty underspecced API so it's not very clear when we should consider the connection ready for use, but I like the idea of confirming that a peering exists just in case.

Reply:
> Is having Peering populated enough to declare that the connection is ready for use?

We may need to use the compute API to list the peerings for the affected GCP network and confirm that the relevant peering is in state `ACTIVE`. I'll check tomorrow to see whether we need to do this.